src/main/java/io/axoniq/axonserver/connector/event/impl/BufferedPersistentStreamSegment.java

```diff
@@ -0,0 +1,130 @@
+package io.axoniq.axonserver.connector.event.impl;
+
+import io.axoniq.axonserver.connector.event.PersistentStreamSegment;
+import io.axoniq.axonserver.connector.impl.StreamClosedException;
+import io.axoniq.axonserver.grpc.event.EventWithToken;
+
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.CopyOnWriteArraySet;
+import java.util.concurrent.LinkedBlockingQueue;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.LongConsumer;
+
+public class BufferedPersistentStreamSegment
```

Comment:
Is there a specific reason this class doesn't extend the `AbstractBufferedStream`?